File: src/org/stormgears/utils/dsio/ISwitch.java
package org.stormgears.utils.dsio;

import edu.wpi.first.wpilibj.command.Command;
import org.jetbrains.annotations.NotNull;

public interface ISwitch {
	void whenFlipped(@NotNull SwitchControl.FlipListener listener);

	void whenPressed(final Command command);

	void whileHeld(final Command command);

	void whenReleased(final Command command);

	void toggleWhenPressed(final Command command);

	void cancelWhenPressed(final Command command);

	void whenActive(final Command command);

	void whileActive(final Command command);

	void whenInactive(final Command command);

	void toggleWhenActive(final Command command);

	void cancelWhenActive(final Command command);
}

File: src/main/java/org/stormgears/utils/dsio/ISwitch.kt
package org.stormgears.utils.dsio

import edu.wpi.first.wpilibj.command.Command

interface ISwitch {
	fun whenFlipped(listener: FlipListener)
	fun whenFlipped(listener: (Boolean) -> Unit)

	fun whenPressed(command: Command)

	fun whileHeld(command: Command)

	fun whenReleased(command: Command)

	fun toggleWhenPressed(command: Command)

	fun cancelWhenPressed(command: Command)

	fun whenActive(command: Command)

	fun whileActive(command: Command)

	fun whenInactive(command: Command)

	fun toggleWhenActive(command: Command)

	fun cancelWhenActive(command: Command)

	fun get(): Boolean

	@FunctionalInterface
	interface FlipListener {
		fun flipped(on: Boolean)
	}
}
